src/jsTest/kotlin/it/krzeminski/githubactionstyping/AjvSchemaValidationTest.kt

```diff
@@ -0,0 +1,100 @@
+package it.krzeminski.githubactionstyping
+
+import ajv.Ajv
+import ajv.ValidateFunction
+import io.kotest.assertions.withClue
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
+import io.kotest.matchers.Matcher
+import io.kotest.matchers.MatcherResult
+import io.kotest.matchers.booleans.shouldBeTrue
+import io.kotest.matchers.nulls.shouldNotBeNull
+import io.kotest.matchers.should
+import io.kotest.matchers.shouldNot
+import io.kotest.mpp.env
+import node.buffer.BufferEncoding.Companion.utf8
+import node.fs.exists
+import node.fs.existsSync
+import node.fs.opendir
+import node.fs.readFile
+import node.fs.readdirSync
+import node.fs.stat
+import node.path.path
+import ajv.Options as AjvOptions
+
+private val schemaFile = env("schemaFile")!!
+private val catalogDir = env("catalogDir")!!
+private val externalDir = env("externalDir")!!
+private val goodDir = env("goodDir")!!
+private val badDir = env("badDir")!!
+
+private lateinit var validate: ValidateFunction
+
+/**
+ * Validate good and bad typings against the Ajv schema validator.
+ */
+class AjvSchemaValidationTest : FunSpec({
+    beforeSpec {
+        withClue("catalogDir should be a non-empty directory") {
+            exists(catalogDir).shouldBeTrue()
+            stat(catalogDir).isDirectory().shouldBeTrue()
+            opendir(catalogDir).read().await().shouldNotBeNull()
+        }
+
+        validate = Ajv(AjvOptions(strict = true)).compile(
+            JSON.parse(readFile(schemaFile, utf8))
+        )
+    }
+
+    if (existsSync(catalogDir)) {
```

Comment:
Why is there the extra check for existence of the dir with the catalog typings? I see these in an extended form in L38-L42.

Reply:
38-42 check it at test execution time and properly report the problem in a meaningful way.

But if we do not check here, we do not reach that code as first this code is executed and the `readdirSync` would fail to register the tests.

Reply:
Ah, got it!